File: tlx toki/re-view/Chapter_01/A_seleksi_olimpiade.cpp
#include <bits/stdc++.h>
using namespace std;

struct nilai{
    string id;
    int s1, s2, s3;
};

nilai n[80];
int N, M;

void swp(nilai &x, nilai &y){
    nilai t = x;
    x = y;
    y = t;
}

void sortt(){
    for(int i = 0; i < M; i++){
        for(int j = i+1; j < N; j++){
            if(n[i].s3 < n[j].s3) swp(n[i], n[j]);
            else if(n[i].s3 == n[j].s3){
                if(n[i].s2 < n[j].s2) swp(n[i], n[j]);
                else if(n[i].s2 == n[j].s2){
                    if(n[i].s1 < n[j].s1) swp(n[i], n[j]);
                }
            }
        }
    }
}

void solve(){
    string s;

    cin >> N >> M >> s;

    for(int i = 0; i < N; i++){
        cin >> n[i].id >> n[i].s1 >> n[i].s2 >> n[i].s3;
    }

    sortt();

    bool res = false;
    for(int i = 0; i < M; i++){
        if(n[i].id == s){
            res = true;
            break;
        }
    }

    if(res){
        cout << "YA" << endl;
    }else{
        cout << "TIDAK" << endl;
    }
}

int main(){
    int t;
    cin >> t;

    while(t--){
        solve();
    }
}

File: tlx toki/re-view/Chapter_01/A_seleksi_olimpiade.py
def swp(arr, a, b):
    arr[a], arr[b] = arr[b], arr[a]

def srt(arr, m):
    for i in range(len(arr)-1):
        for j in range(i+1, len(arr)):
            if arr[i][3] < arr[j][3]:
                swp(arr, i, j)
            elif arr[i][3] == arr[j][3]:
                if arr[i][2] < arr[j][2]:
                    swp(arr, i, j)
                elif arr[i][2] == arr[j][2]:
                    if arr[i][1] < arr[j][1]:
                        swp(arr, i, j)

t = int(input())

    
for _ in range(t):
    n, m = map(int, input().split())
    s = input()

    arr = []
    for i in range(n):
        temp = input().split()
        arr.append([temp[0]] + list(map(int, temp[1:])))

    srt(arr, m)

    res = False
    for i in range(m):
        if arr[i][0] == s:
            res = True
            break
    
    if res:
        print("YA")
    else:
        print("TIDAK")